umapi_client/functional.py

```diff
@@ -195,15 +164,14 @@ def update(self, email=None, username=None, first_name=None, last_name=None, cou
         :param country: new country for this user
         :return: the User, so you can do User(...).update(...).add_to_groups(...)
         """
-        if email:
-            self._validate(email=email)
-        if username:
-            self._validate(username=username)
+        if self.id_type != IdentityTypes.federatedID and email != username:
```

Comment:
This doesn't do what you want.  Suppose they specify just email and no username (which I would expect): this will fail.  Similarly, suppose the case of email and username are different (which is often true): this will fail. I think you mean
```python
if self.id_type != IdentityTypes.federatedID and email and username and email.lower() != username.lower():
```

Reply:
I'll update that.

Reply:
Actually, I'm not we sure should have any checks here.  I updated the condition as per your recommendation and it made [this test](https://github.com/adobe-apiplatform/umapi-client.py/blob/master/tests/test_functional.py#L204) stop working.  The check I had there made it work, but not for the expected reasons.  The root of the issue is that UserAction is semi-stateful - some fields are saved to attributes, but not checked when they should be.  So we either need to remove this check and rely on the UMAPI to tell us if it's wrong, or make sure we manage object state consistently over all method calls (create, update, etc).  Thoughts?

Reply:
Yeah, I see your point.  I think we are better off going with no saved email or domain attributes in the action object, and then doing actual error checking only on the arguments in each UMAPI call.  So in this particular call we would have:
```py
if username and self.id_type != IdentityType.federatedID:
    raise ArgumentError("You cannot set username except for a federated ID")
if email and username and email.lower() == username.lower():
    raise ArgumentError("Specify just email to set both email and username for a federated ID")
```

Reply:
We should also check any other places where we use self.email and self.domain and get rid of them.